AutomatedTesting/cmake/EngineFinder.cmake

```diff
@@ -7,35 +7,72 @@
 #
 #
 # {END_LICENSE}
-# This file is copied during engine registration. Edits to this file will be lost next
-# time a registration happens.
+# Edits to this file may be lost in upgrades. Instead of changing this file, use 
+# the 'engine_finder_cmake_path' key in your project.json or user/project.json to specify 
+# an alternate .cmake file to use instead of this one.
 
 include_guard()
 
-# Read the engine name from the project_json file
+include(cmake/VersionUtils.cmake)
+
+set(ENGINE_FINDER_VERBOSE FALSE CACHE BOOL "Increase EngineFinder.cmake verbosity.  Useful for debugging engine selection based on compatibility.")
+
+# Read the engine name from the shared or user project.json
 file(READ ${CMAKE_CURRENT_SOURCE_DIR}/project.json project_json)
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/project.json)
 
-string(JSON LY_ENGINE_NAME_TO_USE ERROR_VARIABLE json_error GET ${project_json} engine)
+string(JSON O3DE_ENGINE_NAME_TO_USE ERROR_VARIABLE json_error GET ${project_json} engine)
 if(json_error)
     message(FATAL_ERROR "Unable to read key 'engine' from 'project.json'\nError: ${json_error}")
 endif()
+ 
+# Let the user override the engine to use
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/user/project.json)
+    file(READ ${CMAKE_CURRENT_SOURCE_DIR}/user/project.json user_project_json)
+    set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/user/project.json)
```

Comment:
This only needs to be set as a CMake configure dependency if it has the "engine" key within it.

If the "engine" key is not within the `user/project.json` file, then CMake doesn't need to re-run configure when a generator project builds.